source/common/http/codec_client.cc

```diff
@@ -12,8 +12,10 @@ namespace Envoy {
 namespace Http {
 
 CodecClient::CodecClient(Type type, Network::ClientConnectionPtr&& connection,
```

Comment:
I think there may be places in which codec clients get destroyed as part of deferred deletion. Can you make sure the idle timer is disabled/destroyed when the codec client is closed by any mechanism? Please add a test that covers this also (close client when timer is active and no active requests).